assets/js/blocks/collection-filters/inner-blocks/attribute-filter/edit.tsx

```diff
@@ -0,0 +1,155 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useEffect, useState } from '@wordpress/element';
+import { BlockControls, useBlockProps } from '@wordpress/block-editor';
+import { getSetting } from '@woocommerce/settings';
+import { useCollection } from '@woocommerce/base-context/hooks';
+import { AttributeSetting, AttributeTerm } from '@woocommerce/types';
+import {
+	Disabled,
+	Button,
+	ToolbarGroup,
+	withSpokenMessages,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import type { EditProps } from './types';
+import {
+	NoAttributesPlaceholder,
+	AttributesPlaceholder,
+} from './components/placeholder';
+import { AttributeSelectControls } from './components/attribute-select-controls';
+import { getAttributeFromId, isDeepEqual } from './utils';
+import { Inspector } from './components/inspector-controls';
+import { AttributeCheckboxList } from './components/attribute-checkbox-list';
+import { AttributeDropdown } from './components/attribute-dropdown';
+import './style.scss';
+
+const ATTRIBUTES = getSetting< AttributeSetting[] >( 'attributes', [] );
+
+const Edit = ( props: EditProps ) => {
+	const {
+		attributes: blockAttributes,
+		setAttributes,
+		debouncedSpeak,
+	} = props;
+
+	const {
+		attributeId,
+		queryParam,
+		queryType,
+		isPreview,
+		displayStyle,
+		showCounts,
+	} = blockAttributes;
+
+	const attributeObject = getAttributeFromId( attributeId );
+
+	const [ isEditing, setIsEditing ] = useState(
+		! attributeId && ! isPreview
+	);
+
+	const { results: attributeTerms } = useCollection< AttributeTerm >( {
+		namespace: '/wc/store/v1',
+		resourceName: 'products/attributes/terms',
+		resourceValues: [ attributeObject?.id || 0 ],
+		shouldSelect: blockAttributes.attributeId > 0,
+		query: { orderby: 'menu_order' },
+	} );
+
+	const blockProps = useBlockProps();
+
+	useEffect( () => {
+		if ( ! attributeObject?.taxonomy ) {
+			return;
+		}
+		const newQueryParam = {
```

Comment:
This one certainly raises my code-smell sense a little with the deepEqual. For one thing in the impl of deep equal it does a `===` shortcut but that will never happen, because you always construct a new ref for newQueryParam,

Is the structure of queryParam and newQueryParam always the same? just wondering if this could/should just be:

```
const {calculate_attribute_counts: { taxonomy: oldTaxonomy, queryType: oldQueryType }} = queryParam;

if ( oldTaxonomy === attributeObject.taxonomy && oldQueryType === queryType ) {
  return;
}
```

I'll admit though its not more elegant or anything, it just felt like the introduction of deepEqual felt a little redundant.

I also wonder for something as generically useful as deepEqual if we do introduce it, can we introduce it to a more central lib of utilities? Seems a bit off to tuck it away here in a block. I'd be fine with either approach here personally.

Reply:
@samueljseay You're right. The `isDeepEqual` was introduced to fix duplicated data when I used array for attribute query param. Later on, I converted it to object, so the utility became unnecessary. I remove it in 041fce3.